### PR TITLE
GEODE-3741: removed debug statements and cleanup

### PR DESCRIPTION
**Pull request by mhansonp**
- Changed some definitions to auto
- Removed some debug statements

- note: No shared_ptr change or loop change as they are not necessary. 
With regard to the "for" loop would not work, hence the while loop.

### CODE REVIEW

**Comment by pivotal-jbarrett** [comment]
@mhansonp Please explain why there are not necessary. The use of delete is strongly discouraged in C++11 and our standards is to convert to C++11. We should be cleaning up things like this.

Please explain why for loop does not work?

**Comment by mhansonp** [comment]
Regarding the loop we are actively changing the size of the vector by erasing entries, which makes using the iterator troublesome; i.e. the container is changed after the iterator is initialized.

Regarding the shared_ptr, the vector is created and destroyed in this function. The entries are properly deleted as well, switching to a shared_ptr for something fully contained within a function is heavy handed.

**Comment by dgkimura** [comment]
I too strongly prefer to use smart pointers instead of managing memory ourselves.  But if everything is local to this function, why do we even need to do a heap allocation at all?

**Comment by pivotal-jbarrett** [comment]
@dgkimura I think the elements are filled in from another function, though that function could possibly use values. I am good with either values or shared_ptr. Not happy with delete!

**Comment by pivotal-jbarrett** [comment]
@echobravopapa Editing comments to respond to them does not result in an alert.

File: cppcache/src/ThinClientRegion.cpp [hunk 1]
```diff
@@ -3255,25 +3181,24 @@ bool ThinClientRegion::executeFunctionSH(
...
   for (const auto& locationIter : *locationMap) {
     const auto& serverLocation = locationIter.first;
     const auto& routingObj = locationIter.second;
-    auto worker = new OnRegionFunctionExecution(
+    auto worker = std::make_shared<OnRegionFunctionExecution>(
         func, this, args, routingObj, getResult, timeout,
         dynamic_cast<ThinClientPoolDM*>(m_tcrdm), resultCollectorLock, rc,
         userAttr, false, serverLocation, allBuckets);
-    threadPool->perform(worker);
+    threadPool->perform(worker.get());
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply cppcache/src/ThinClientRegion.cpp]: I actually disagree. I don't like the idea of passing in a shared_ptr, I understand that eventually that may change, but for the moment it is misleading in a way and I think that is a bad thing. I think if we made a change away from the underlying ace code that made this actually use the shared_ptr, then we can refactor then and the use will be obvious and correct.

File: cppcache/src/ThinClientRegion.cpp [hunk 2]
```diff
@@ -3255,25 +3181,24 @@ bool ThinClientRegion::executeFunctionSH(
...
     feWorkers.push_back(worker);
   }
 
   GfErrType abortError = GF_NOERR;
 
-  while ( !feWorkers.empty()) {
-    auto iter = feWorkers.begin();
+  for ( auto iter = std::begin(feWorkers); iter != std::end(feWorkers);) {
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply cppcache/src/ThinClientRegion.cpp]: ok, done.

**Pull request merged**